drivers/timer/ite_it8xxx2_timer.c

```diff
@@ -122,6 +137,32 @@ void timer_5ms_one_shot(void)
 }
 #endif /* CONFIG_SOC_IT8XXX2_PLL_FLASH_48M */
 
+#ifdef CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT
+__timer_ram_code void arch_busy_wait(uint32_t usec_to_wait)
+{
+	if (!usec_to_wait) {
+		return;
+	}
+
+	/* Decrease 1us here to calibrate our access registers latency */
+	usec_to_wait--;
+
+	/*
+	 * We want to set the bit(1) re-start busy wait timer soon as
+	 * possible, so we directly write 0xb instead of | bit(1).
+	 */
+	IT8XXX2_EXT_CTRLX(BUSY_WAIT_L_TIMER) = IT8XXX2_EXT_ETX_COMB_RST_EN;
+
+	for (;;) {
+		uint32_t curr = IT8XXX2_EXT_CNTOX(BUSY_WAIT_H_TIMER);
+
+		if (curr >= usec_to_wait) {
+			break;
+		}
+	}
```

Comment:
Is there a reason to do `for (;;)` here instead of `do {} while`? Seems better suited for this code.

Reply:
I compare the assembly code for for(;;) and do{}while loop in below:


for loop
800044aa <arch_busy_wait>:
	if (!usec_to_wait) {
800044aa:	cd09                	beqz	a0,800044c4 <arch_busy_wait+0x1a>
	IT8XXX2_EXT_CTRLX(BUSY_WAIT_L_TIMER) = IT8XXX2_EXT_ETX_COMB_RST_EN;
800044ac:	472d                	li	a4,11
800044ae:	00f027b7          	lui	a5,0xf02
800044b2:	f2e78023          	sb	a4,-224(a5) # f01f20 <CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC+0xef9f20>
	usec_to_wait--;
800044b6:	157d                	addi	a0,a0,-1
		uint32_t curr = IT8XXX2_EXT_CNTOX(BUSY_WAIT_H_TIMER);
800044b8:	00f02737          	lui	a4,0xf02
800044bc:	f5472783          	lw	a5,-172(a4) # f01f54 <CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC+0xef9f54>
		if (curr >= usec_to_wait) {
800044c0:	fea7eee3          	bltu	a5,a0,800044bc <arch_busy_wait+0x12>
}
800044c4:	8082                	ret


uint32_t curr;
do {
	curr = IT8XXX2_EXT_CNTOX(BUSY_WAIT_H_TIMER);
} while (curr < usec_to_wait);
800044aa <arch_busy_wait>:
	if (!usec_to_wait) {
800044aa:	cd09                	beqz	a0,800044c4 <arch_busy_wait+0x1a>
	IT8XXX2_EXT_CTRLX(BUSY_WAIT_L_TIMER) = IT8XXX2_EXT_ETX_COMB_RST_EN;
800044ac:	472d                	li	a4,11
800044ae:	00f027b7          	lui	a5,0xf02
800044b2:	f2e78023          	sb	a4,-224(a5) # f01f20 <CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC+0xef9f20>
	usec_to_wait--;
800044b6:	157d                	addi	a0,a0,-1
		curr = IT8XXX2_EXT_CNTOX(BUSY_WAIT_H_TIMER);
800044b8:	00f02737          	lui	a4,0xf02
800044bc:	f5472783          	lw	a5,-172(a4) # f01f54 <CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC+0xef9f54>
	} while (curr < usec_to_wait);
800044c0:	fea7eee3          	bltu	a5,a0,800044bc <arch_busy_wait+0x12>
}
800044c4:	8082                	ret


I think both have the same efficiency in the loop, should I change to do{}while?